tabbycat/results/forms.py

```diff
@@ -488,6 +494,9 @@ def create_participant_fields(self):
             self.fields[self._fieldname_speaker(side, pos)] = forms.ModelChoiceField(
                 queryset=queryset, required=True)
 
+            if self.tournament.pref('speaker_ranks') != 'none':
+                self.fields[self._fieldname_srank(side, pos)] = forms.IntegerField(required=True, min_value=1, max_value=nspeeches, step_size=1)
+
```

Comment:
I think these would be better thought of as score fields, and the logic moved to `SingleBallotSetForm.create_score_fields` and (possibly) `PerAdjudicatorBallotSetForm.create_score_fields`.

I guess the reason `self.create_score_fields()` is extract to its own method is because the score fields differ depending on whether it's voting or consensus ballots. I think in principle this is also true for speaker ranks—if adjudicators are scoring speakers independently, they should also be ranking speakers independently(?)

If APDA never does voting ballots, I'm fine with only implementing the logic for consensus ballots, and ignoring the speaker ranks setting in voting ballots (_i.e._ only putting the logic in `SingleBallotSetForm`). As things currently stand, when it uses voting ballots, our form logic creates only one set of speaker rank fields, but renders those same fields once per adjudicator, which leads to weird behaviours caused by duplicate fields.

<img width="911" alt="image" src="https://user-images.githubusercontent.com/1725499/230750381-68f56626-f7ed-4057-a586-d10c6aaebe08.png">